src/main/java/com/scriptbasic/executors/commands/CommandSelect.java

```diff
@@ -0,0 +1,68 @@
+package com.scriptbasic.executors.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.scriptbasic.api.ScriptBasicException;
+import com.scriptbasic.interfaces.Expression;
+import com.scriptbasic.spi.Interpreter;
+
+public class CommandSelect extends AbstractCommandSelectPart {
+	
+	static public final String CASE_APPLIED = "SELECT_CASE_APPLIED";
+	
+	/**
+	 * Expression in select statement
+	 */
+	private Expression expression;
+
+	private CommandEndSelect commandEndSelect;
+	
+	/**
+	 * List of possible cases
+	 */
+	private List<CommandCase> cases = new ArrayList<>();
+
+	public void setExpression(final Expression expression) {
+		this.expression = expression;		
+	}	
+
+	@Override
+	public void execute(Interpreter interpreter) throws ScriptBasicException {
+		// Evaluate expression
+		final var expressionValue = expression.evaluate(interpreter);
+		// Mark as not finished
+		interpreter.getMap().put(CASE_APPLIED, false);
+		
+		// Iterate all cases and calls first matching cases

```

Comment:
Iterate all cases and calls first matching cases ->

Iterate over all the cases and sets the next command pointer/index on the first matching case